.github/workflows/protect-readmes.yml

```diff
@@ -0,0 +1,20 @@
+name: Protect generated developer guide
+# Ensure that READMEs are not inadvertently changed in a pull request as those changes
+# would be dropped on the next CI run
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  protect-developer-guide:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
```

Comment:
I think we need a bigger fetch depth here, the default one is 1, so I think that the `main` branch doesn't have to actually get pulled.

Reply:
Yup, just realized and already force-pushed before reading :innocent: